### PR TITLE
api: add `Untyped` metadata support in ExtProc

### PR DESCRIPTION
**Pull request by zirain**
xref: #4152

### CODE REVIEW

File: api/v1alpha1/ext_proc_types.go [hunk 1]
```diff
@@ -70,4 +70,16 @@ type ExtProc struct {
...
+
+	// MetadataOptions defines the metadata that should be forwarded to or received from the external processor.
+	//
+	// +optional
+	Metadata *MetadataOptions `json:"metadata,omitempty"`
+}
+
+type MetadataOptions struct {
+	// ForwardingNamespaces is a list of namespaces from which metadata should be forwarded to the external processor.
+	ForwardingNamespaces []string `json:"forwardingNamespaces,omitempty"`
```
Comment (guydc) [inline api/v1alpha1/ext_proc_types.go:82]: Will this be translated to typed or untyped namespaces? if we want to support both options in the future, we maybe want to make the distinction here already, like here: https://github.com/envoyproxy/gateway/pull/3247/files#diff-2d475e566eff6abe27d7ed2892b3dd252980f2cfccc04bb859a14878c8ca71b1R63

Comment (zirain) [reply api/v1alpha1/ext_proc_types.go:82]: can you resubmit that one instead of this?

Comment (guydc) [reply api/v1alpha1/ext_proc_types.go:82]: it's stale, let's just go ahead with your PR.

**Review by guydc — APPROVED** [review]
LGTM

**Review by Xunzhuo — APPROVED** [review]
/lgtm

**Pull request closed without merging**